mathics/builtin/assignments/internals.py

```diff
@@ -153,7 +154,7 @@ def find_tag_and_check(lhs, tags, evaluation):
 
 
 def unroll_patterns(lhs, rhs, evaluation):
-    if type(lhs) is Symbol:
+    if isinstance(lhs, Atom):
```

Comment:
Is the idea that unrolling works if we have number as well and we weren't catching this?

Reply:
This is something that found by chance: suppose that by mistake, you evaluate
`7/; F[a]:=7`
in master, it raises an unhandled exception. With this change, just an error message.